enterprise_gateway/services/processproxies/processproxy.py

```diff
@@ -583,31 +579,58 @@ def _get_ssh_client(self, host):
         """
         ssh = None
 
-        global remote_user
-        global remote_pwd
-        if remote_user is None:
-            remote_user = os.getenv('EG_REMOTE_USER', getpass.getuser())
-            remote_pwd = os.getenv('EG_REMOTE_PWD')  # this should use password-less ssh
+        use_gss = os.getenv("EG_REMOTE_GSS_SSH", None)
+        if self.remote_user is None:
+            self.remote_pwd = os.getenv(
+                "EG_REMOTE_PWD"
+            )  # this should use password-less ssh
+            self.remote_user = os.getenv("EG_REMOTE_USER", getpass.getuser())
+
+            if use_gss and (self.remote_pwd or self.remote_user):
```

Comment:
As you are defaulting `remote_user` to `getpass.getuser()` then when `use_gss` is True it seems this will always print the warning even if neither `EG_REMOTE_PWD` or `EG_REMOTE_USER` have been set.

Reply:
i've move the logic for the warning to `__init__` that might not be exactly the same but I doubt there are use case where use change the value of env variable after the object have been constructed.

That should take care of the problem.